.github/workflows/ci-cpu-cpp.yml

```diff
@@ -16,17 +16,42 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-20.04, macOS-latest]
+        os: [ubuntu-20.04]
     steps:
+      # - name: Setup Python for M1
+      #   if: matrix.os == 'macos-14'
+      #   uses: actions/setup-python@v5
+      #   with:
+      #     python-version: '3.10'
+      - name: Setup Python for all other OS
+        if: matrix.os != 'macos-14'
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9
+          architecture: x64
+      - name: Setup Java 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: '17'
       - name: Checkout TorchServe
-        uses: actions/checkout@v2
-      - name: Install libtorch - macOS
-        if: matrix.os == 'macOS-latest'
-        run: |
-            brew install libtorch
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+      # - name: Install libtorch - macOS
+      #   if: matrix.os == 'macOS-latest'
+      #   run: |
+      #       brew install libtorch
       - name: Install dependencies
         run: |
-          python ts_scripts/install_dependencies.py --environment=dev --cpp
+          sudo apt update && python ts_scripts/install_dependencies.py --environment=dev --cpp
```

Comment:
can we check cmake version before we proceed?

Reply:
@lxning not sure what you mean? Inside the workflow file? What would be the consequences of the check?

Reply:
install_dependency with option "cpp" will install cmake (eg. [mac](https://github.com/pytorch/serve/blob/master/ts_scripts/install_dependencies.py#L56), [linux](https://github.com/pytorch/serve/blob/master/ts_scripts/install_dependencies.py#L17)). Currently some cmake version can cause cpp compile error. Better to check cmake version before cijob proceed to save debug time.